### ISSUE TITLE
Loading non-compiled package

### ISSUE THREAD

**Issue by olexandr-konovalov** [gap-packages/curlInterface#15]
I have a 32bit build in which compiling fails:
```
checking for GAP architecture... x86_64-pc-linux-gnu-default32
checking for GAP >= 4.9... yes
checking for gawk... (cached) gawk
checking for curl-config... /usr/bin/curl-config
checking for the version of libcurl... 7.29.0
checking whether libcurl is usable... no
configure: error: libcurl development files required
Running 'gmake' 
gmake: *** No targets specified and no makefile found.  Stop.

WARNING: Failed to build curlInterface-2.0.2

Tue 11 Sep 02:59:11 BST 2018
```
The package however still loads, and then tests have multiple problems:
```
============================OUTPUT START==============================
#I  RunPackageTests("curlinterface", "2.0.2", "tst/testall.g", "false");
 ���������������������������   GAP 4.dev of today
 ���  GAP  ���   https://www.gap-system.org
 ���������������������������   Architecture: x86_64-pc-linux-gnu-default32
 Configuration:  gmp 6.1.2, readline
 Loaded workspace: wsp.g
 Packages:   curlInterface 2.0.2, GAPDoc 1.6.1, PrimGrp 3.3.1, SmallGrp 1.3, 
             TransGrp 2.0.4
 Try '??help' for help. See also '?copyright', '?cite' and '?authors'
Architecture: x86_64-pc-linux-gnu-default32

testing: /circa/scratch/gap-jenkins/workspace/GAP-master-test/GAPCOPTS/32build\
/GAPTARGET/packages/label/kovacs/GAP-master-snapshot/pkg/curlInterface-2.0.2/t\
st/basic.tst
########> Diff in /circa/scratch/gap-jenkins/workspace/GAP-master-test/GAPCOPT\
S/32build/GAPTARGET/packages/label/kovacs/GAP-master-snapshot/pkg/curlInterfac\
e-2.0.2/tst/basic.tst:5
# Input is:
r := DownloadURL("http://www.google.com");;
# Expected output:
# But found:
Error, Variable: 'CURL_REQUEST' must have an assigned value
########
```
One should use `AvailabilityTest` (see e.g. https://github.com/gap-packages/example/blob/master/PackageInfo.g) to check whether the binaries are available or not.

**Comment by ChrisJefferson** [comment]
We need something like:

```
AvailabilityTest := function()
  if Filename(DirectoriesPackagePrograms("io"), "io.so") = fail then
    Info(InfoWarning, 1, "CurlInterface: Kernel extension not compiled");
    return fail;
  fi;
  return true;
end,
```

(I haven't even tried running this code, could be wrong).

**Comment by mtorpey** [comment]
PR #16 implements approximately Chris's suggestion.

**Comment by olexandr-konovalov** [comment]
Closed in #16.
